src/main/scala/devices/tilelink/Mtimer.scala

```diff
@@ -0,0 +1,105 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.devices.tilelink
+
+import chisel3._
+import chisel3.util.ShiftRegister
+import org.chipsalliance.cde.config.{Field, Parameters}
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.interrupts._
+import freechips.rocketchip.regmapper._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util._
+
+object MTIMERConsts
+{
+  def mtimecmpOffset(hart: Int) = hart * mtimecmpBytes
+  def mtimecmpBytes   = 8
+  def mtimeWidth      = 64
+  def mtimecmpSize    = 0x8000
+  def mtimeSize       = 0x100
+  def ints            = 1
+}
+
+// Notice: Remember to ensure that the size and address meet the requirements of the AddressSet, if you use ACLINT 
+case class MTIMERParams(MTIMECMPBaseAddress: BigInt = 0x02008000, MTIMEBaseAddress: BigInt = 0x02010000, intStages: Int = 0)
+{
+  def mtimecmpAddress = AddressSet(MTIMECMPBaseAddress, MTIMERConsts.mtimecmpSize - 1)
+
+  def mtimeAddress = AddressSet(MTIMEBaseAddress, MTIMERConsts.mtimeSize - 1)
+}
+
+case class MTIMERAttachParams(
+  slaveWhere: TLBusWrapperLocation = CBUS
+)
+
+case object MTIMERAttachKey extends Field(MTIMERAttachParams())
+
+class MTIMER(params: MTIMERParams, beatBytes: Int)(implicit p: Parameters) extends LazyModule
+{
+  import MTIMERConsts._ 
+
+  val device = new SimpleDevice("mtimer", Seq("riscv,aclint-mtimer")) {
```

Comment:
mtimer exposes `reg = <0x2008000 0x8100>;` to the dts, which does not show where mtimer base address is. We might see how opensbi find mtimer base address

https://github.com/riscv-software-src/opensbi/blob/dc1c7db05e075e0910b93504370b50d064a51402/lib/utils/timer/fdt_timer_mtimer.c#L71-L74

Seems like it should be something like `reg = <0x2008000 0x7FF0 0x2010000 0x8>;`